src/MicrosoftTeams.ts

```diff
@@ -182,6 +387,16 @@ namespace microsoftTeams {
   let backButtonPressHandler: () => boolean;
   handlers["backButtonPress"] = handleBackButtonPress;
 
+  handlers["nativeClientInitialize"] = handleNativeClientInitialize;
```

Comment:
What is this needed for? Initialization should always happen in response to the SDK’s initalize call; not in a handler.

Reply:
In a iframe-less scenario, Android client app needs to inject js into the page to receive even the first "initialize" message. This handler is invoked when the client app is done with injection and is ready to receive the messages. Till that time the messages are held in a queue in the SDK.

Reply:
But you have no guarantee that the loaded page has even initialized the SDK. More importantly, they might be using it via the npm package at which point it would just be part of their app bundle and not in the global namespace. The SDK has to always be the one that hooks up the message handler to some global object (e.g. the Window). Your injected code must not make direct calls to the SDK; it must only post messages via the registered handler.

Reply:
I didn't get you here. Can you share some code snippet of what you have in mind?

Reply:
It seems like your plan is to have the injected code call microsofTeams.native.postMessage in order to send messages to the SDK. What I am trying to say is that this will not work in cases where the microsoftTeams object isn't available in the global namespace. That could happen because the <script> tag that pulls in the SDK is being delay loaded or the developer has bundled the SDK with their app's code.

You must instead follow the pattern that was established by the iOS version of the SDK that Assignments uses, i.e. register a handler on some global object when the developer calls microsoftTeams.initialize(). That would look something like:
```
window.onNativeMessage = handleParentMessage;
```

Then in the injected code you just call onNativeMessage when you want to post something to the SDK.

Reply:
I don't see you registering  window.microsoftTeams.native.postMessage anywhere. And yes, please remove this namespace and method from the SDK and don't export it. All you need to do is register the handler inside of microsoftTeams.initialize. I am going to repeat what I said earlier; you **cannot** rely on the microsoftTeams object to be globally available on the window; you must explicitly register the global handler at runtime when the hosted web app calls microsoftTeams.initialize().

Reply:
If you want to see what I mean just build a web app using typescript or webpack+babel and use the SDK via its npm package. In that case you wouldn't reference the SDK from our CDN but instead you would first:
```
yarn add @microsoft/teams-js
```

And then inside of your code you would do something like:
```
import * as microsoftTeams from '@microsoft/teams-js';
microsoftTeams.initialize();
```

For an app built in this way window.microsoftTeams would always be undefined.